planner/core/logical_plan_builder.go

```diff
@@ -3841,6 +3842,26 @@ func (b *PlanBuilder) buildDataSource(ctx context.Context, tn *ast.TableName, as
 		// Try CTE.
 		p, err := b.tryBuildCTE(ctx, tn, asName)
 		if err != nil || p != nil {
+			if len(b.cteNameProcessingStack) > 0 {
+				if tblIds, ok := b.cteToTblIDMapper[tn.Name.L]; ok {
+					cteName := b.cteNameProcessingStack[len(b.cteNameProcessingStack)-1]
+					b.cteToTblIDMapper[cteName] = append(b.cteToTblIDMapper[cteName], tblIds...)
+				}
+			} else {
+				if tblIds, ok := b.cteToTblIDMapper[tn.Name.L]; ok {
```

Comment:
it seems like this line is duplicated with 3846, could we simplify the code?

Reply:
done